sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -502,10 +502,14 @@ object SQLConf {
   val COLUMN_VECTOR_OFFHEAP_ENABLED =
     buildConf("spark.sql.columnVector.offheap.enabled")
       .internal()
-      .doc("When true, use OffHeapColumnVector in ColumnarBatch.")
+      .doc("When true, use OffHeapColumnVector in ColumnarBatch. " +
+        s"Defaults to $MEMORY_OFFHEAP_ENABLED && $MEMORY_OFFHEAP_SIZE > 0.")
       .version("2.3.0")
       .booleanConf
-      .createWithDefault(false)
+      .createWithDefaultFunction(() => {
```

Comment:
I think you can use `fallbackConf`

Reply:
Updated. But now the off-heap size check is removed, I hope it's fine.